vars/highLevelDataSetup.groovy

```diff
@@ -13,7 +13,7 @@ def call(params) {
 
   if (config.highLevelDataSetup) {
     stageWithAgent("High Level Data Setup - ${environment}", product) {
-      withDefinitionImportSecretsAndEnvVars(product, environment, config.vaultEnvironmentOverrides){
+      withDefinitionImportSecretsAndEnvVars(product, environment, config.highLevelDataSetupKeyVaultName, config.vaultEnvironmentOverrides){
```

Comment:
this is where you should check if the config value is non null and instead send that name instead of the product

Reply:
Rework completed